techSpec.js

```diff
@@ -0,0 +1,209 @@
+/*
+2. Introduction
+
+a. Overview, Problem Description, Summary, or Abstract
+
+    Context: Cat API is an Express.js application run on Node.js runtime. 
+
+    Problem: The Cat API currently does not have access to a cat database that persists. 
+    The information is generated when the API server is started. 
+    For this reason, for example, the ids are different every time around which made writing test cases trickier. 
+
+    Suggested solution: Implement a SQLite database which allows information to be persisted. 
+
+b. Glossary  or Terminology
+
+    SQL - structured query language 
+    SQLite - a file-based database engine that uses SQL syntax
+    sqlite3 - node module enables connection to SQLite databases in memory or disk file 
+
+    sqlite3 methods:
+        db.all - allows us to access all the rows in the dataset - this might offer the best flexibility for now
+        db.each - executes code for each row in a dataset
+        db.get - executes code for the FIRST row - use when we know only one row will return. 
+
+d. Goals or Product and Technical Requirements
+
+    Product requirements: 
+        1. We should have a pre-populated SQLite database on the file system. 
+        2. Upon start-up of the Cat API Express server, it should establish a connection to the SQL database - or should it?  
+           Let's review:
+           The connection is required wherever we make a query via sqlite3 methods: all() get() and each()
+           The connection should be closed when we finish making the query and processing the result set. 
+           Therefore the connection does NOT need to be connected immediately upon server start-up.
+        3. The Cat API should be able to read from the SQLite database for GET all & GET by id requests and return objects. 
+        4. The Cat API should be able to write to the SQLite database for POST & PUT requests and return new/updated objects. 
+        4. The Cat API should be able to delete from the SQLite database for DEL requests. 
+        
+
+    Technical requirements: 
+        
+        1.
+        Create and populate 'cat' & 'pedigree breed' tables via sqlite in Terminal. 'CREATE TABLE' and so forth....
+        Remove list of cats and breed objects from catsRepo.js & breedsRepo.js respectively.   
+        Review if the repositories still need to be classes in this case? 
```

Comment:
Yes, they should be. The advantage of the repository pattern is that your routers shouldn't need a single line of code changing when you swap out an in-memory list for a database on disk. You might want to make similar alterations in future so it's still helpful to retain this abstraction.